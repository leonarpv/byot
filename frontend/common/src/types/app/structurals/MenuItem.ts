import * as React from 'react';

export interface MenuItem<I = string | number> {
  label: string;
  id: I;
  icon?: React.ReactNode;
}

export interface LinkMenuItem<I = string | number> extends MenuItem<I> {
  link: string;
}
