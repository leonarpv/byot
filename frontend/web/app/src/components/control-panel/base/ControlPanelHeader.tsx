import * as React from 'react';
import {Box, Grid, IconButton, InputBase, makeStyles, Theme, useTheme} from '@material-ui/core';
import {Menu, Search} from '@material-ui/icons';
import {useTranslation} from 'react-i18next';
import {WithStyles} from '@byot-frontend/web-common/src/types/WithStyles';
import {Logo} from '@byot-frontend/web-common/src/components/elementary/logo/Logo';
import {UserAvatarDropdown} from '../user/UserAvatarDropdown';

interface Props extends WithStyles<typeof styles> {}

const styles = (theme: Theme) => ({
  root: {
    backgroundColor: theme.palette.common.white,
    position: 'fixed' as 'fixed',
    left: 0,
    top: 0,
    width: '100%',
  },
});
const useStyles = makeStyles(styles);

export const ControlPanelHeader: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  const {t} = useTranslation();
  const theme = useTheme();
  return (
    <Grid container classes={{root: styles.root}} alignItems="center">
      <Grid item>
        <Box py={1} px={2}>
          <IconButton>
            <Menu />
          </IconButton>
        </Box>
      </Grid>
      <Grid item>
        <Box py={1} px={2} display="flex" flexDirection="row" alignItems="center">
          <Search />
          <InputBase placeholder={t('Type in to search...')} />
        </Box>
      </Grid>
      <Grid item md>
        <Box py={1} px={2} textAlign="center">
          <Logo height={theme.spacing(4)} />
        </Box>
      </Grid>
      <Grid item>
        <UserAvatarDropdown />
      </Grid>
    </Grid>
  );
};
