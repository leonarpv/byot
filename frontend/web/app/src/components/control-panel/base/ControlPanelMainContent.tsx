import * as React from 'react';
import {Box} from '@material-ui/core';

interface Props {
  children?: React.ReactNode;
}

export const ControlPanelMainContent: React.FC<Props> = (props: Props) => {
  return (
    <Box px={4} pt={4}>
      {props.children}
    </Box>
  );
};
