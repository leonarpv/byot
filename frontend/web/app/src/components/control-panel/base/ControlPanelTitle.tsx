import * as React from 'react';
import {Box, Typography} from '@material-ui/core';

interface Props {
  children?: React.ReactNode;
}

export const ControlPanelTitle: React.FC<Props> = (props: Props) => {
  return (
    <Box mb={4} mt={2}>
      <Typography variant="h3">{props.children}</Typography>
    </Box>
  );
};
