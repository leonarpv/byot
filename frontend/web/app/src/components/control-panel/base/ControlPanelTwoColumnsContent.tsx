import * as React from 'react';
import {makeStyles, Theme, Grid} from '@material-ui/core';
import {WithStyles} from '@byot-frontend/web-common/src/types/WithStyles';

interface Props extends WithStyles<typeof styles> {
  children?: React.ReactNode;
  secondaryContent?: React.ReactNode;
}

const styles = (theme: Theme) => ({
  mainColumn: {
    [theme.breakpoints.up('md')]: {
      paddingRight: '33%',
    },
  },
  secondaryColumn: {
    padding: theme.spacing(2),
    right: 0,
    top: 0,
    width: '100%',
    [theme.breakpoints.up('md')]: {
      position: 'fixed' as 'fixed',
      width: '25%',
    },
  },
});
const useStyles = makeStyles(styles);

export const ControlPanelTwoColumnsContent: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  return (
    <>
      <div className={styles.mainColumn}>{props.children}</div>
      <div className={styles.secondaryColumn}>{props.secondaryContent}</div>
    </>
  );
};
