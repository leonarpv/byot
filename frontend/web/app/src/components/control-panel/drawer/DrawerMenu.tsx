import * as React from 'react';
import {makeStyles, Theme, List} from '@material-ui/core';
import {Home} from '@material-ui/icons';
import {WithStyles} from '@byot-frontend/web-common/src/types/WithStyles';
import {DrawerMenuItem} from './DrawerMenuItem';

interface Props extends WithStyles<typeof styles> {}

const styles = (theme: Theme) => ({});
const useStyles = makeStyles(styles);

export const DrawerMenu: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  return (
    <List disablePadding>
      <DrawerMenuItem icon={<Home />} link="/">
        asdlfkjsd
      </DrawerMenuItem>
    </List>
  );
};
