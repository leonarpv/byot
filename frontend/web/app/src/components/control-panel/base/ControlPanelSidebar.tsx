import * as React from 'react';
import {makeStyles, Theme, Box} from '@material-ui/core';
import {ExitToAppRounded} from '@material-ui/icons';
import {WithStyles} from '@byot-frontend/web-common/src/types/WithStyles';
import {DrawerMenu} from '../drawer/DrawerMenu';
import {Logo} from '@byot-frontend/web-common/src/components/elementary/logo/Logo';
import {Copyright} from '@byot-frontend/web-common/src/components/content/Copyright';
import {LinkMenuItem} from '@byot-frontend/common/src/types/app/structurals/MenuItem';
import {useTranslation} from 'react-i18next';
import {Router} from '../../../router/Router';

interface Props extends WithStyles<typeof styles> {
  menuItems: LinkMenuItem[];
  selected?: string;
}

const styles = (theme: Theme) => ({
  root: {
    position: 'fixed' as 'fixed',
    left: 0,
    top: 0,
    height: '100%',
    padding: theme.spacing(2),
    display: 'flex' as 'flex',
    flexDirection: 'column' as 'column',
    justifyContent: 'space-between' as 'space-between',
    alignItems: 'center' as 'center',
    width: '100%',
    [theme.breakpoints.between('sm', 'md')]: {
      width: '33%',
    },
    [theme.breakpoints.up('md')]: {
      width: '25%',
    },
  },
});
const useStyles = makeStyles(styles);

export const ControlPanelSidebar: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  const {t} = useTranslation();
  return (
    <div className={styles.root}>
      <Logo height={64} />
      <DrawerMenu menu={props.menuItems} selected={props.selected} />
      <Box width="100%" textAlign="center">
        <DrawerMenu
          disablePadding
          menu={[
            {
              label: t('Log out'),
              link: Router.logout.URI(),
              id: 'logout',
              icon: <ExitToAppRounded />,
            },
          ]}
        />
        <Copyright />
      </Box>
    </div>
  );
};
