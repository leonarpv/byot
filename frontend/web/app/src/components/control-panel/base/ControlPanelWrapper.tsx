import * as React from 'react';
import {makeStyles, Theme} from '@material-ui/core';
import {WithStyles} from '@byot-frontend/web-common/src/types/WithStyles';
import {ControlPanelSidebar} from './ControlPanelSidebar';
import {LinkMenuItem} from '@byot-frontend/common/src/types/app/structurals/MenuItem';

interface Props extends WithStyles<typeof styles> {
  children: React.ReactNode;
  menuItems: LinkMenuItem[];
  selected?: string;
}

const styles = (theme: Theme) => ({
  content: {
    [theme.breakpoints.between('sm', 'md')]: {
      paddingLeft: '33%',
    },
    [theme.breakpoints.up('md')]: {
      paddingLeft: '25%',
    },
  },
});
const useStyles = makeStyles(styles);

export const ControlPanelWrapper: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  return (
    <>
      <ControlPanelSidebar selected={props.selected} menuItems={props.menuItems} />
      <div className={styles.content}>{props.children}</div>
    </>
  );
};
