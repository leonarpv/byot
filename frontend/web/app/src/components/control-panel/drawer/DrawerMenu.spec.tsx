import * as React from 'react';
import {render} from '@testing-library/react';
import {DrawerMenu} from './DrawerMenu';
import {LinkMenuItem} from '@byot-frontend/common/src/types/app/structurals/MenuItem';
import {ControlPanelWrapper} from '../base/ControlPanelWrapper';

describe('<DrawerMenu/>', () => {
  it('should render', () => {
    const {container} = render(<DrawerMenu menu={[]} />);
    expect(container).toMatchSnapshot();
  });
  it('should render as many items as pr', () => {
    const menuItems: LinkMenuItem[] = [
      {
        id: 0,
        label: 'Label 0',
        link: '/route-0',
      },
      {
        id: 1,
        label: 'Label 1',
        link: '/route-1',
      },
    ];
    const {container, queryAllByTestId} = render(<DrawerMenu menu={menuItems} />);
    expect(queryAllByTestId('control-panel-drawer-menu-item')).toHaveLength(menuItems.length);
    expect(container).toMatchSnapshot();
  });
});
