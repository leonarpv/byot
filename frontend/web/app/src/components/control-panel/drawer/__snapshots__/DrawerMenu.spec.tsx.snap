// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerMenu/> should render 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2 makeStyles-root-1"
  >
    <ul
      class="MuiList-root"
    />
  </div>
</div>
`;

exports[`<DrawerMenu/> should render as many items as pr 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-4 makeStyles-root-3"
  >
    <ul
      class="MuiList-root"
    >
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root makeStyles-root-5 MuiListItem-gutters MuiListItem-button"
        data-testid="control-panel-drawer-menu-item"
        role="button"
        tabindex="0"
        to="/route-0"
      >
        <span
          class="makeStyles-highlight-6 makeStyles-highlight-9"
        />
        <div
          class="MuiListItemIcon-root makeStyles-icon-7 makeStyles-icon-10"
        />
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary makeStyles-textPrimary-8 makeStyles-textPrimary-11 MuiTypography-body1 MuiTypography-displayBlock"
          >
            Label 0
          </span>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root makeStyles-root-5 MuiListItem-gutters MuiListItem-button"
        data-testid="control-panel-drawer-menu-item"
        role="button"
        tabindex="0"
        to="/route-1"
      >
        <span
          class="makeStyles-highlight-6 makeStyles-highlight-12"
        />
        <div
          class="MuiListItemIcon-root makeStyles-icon-7 makeStyles-icon-13"
        />
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary makeStyles-textPrimary-8 makeStyles-textPrimary-14 MuiTypography-body1 MuiTypography-displayBlock"
          >
            Label 1
          </span>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </ul>
  </div>
</div>
`;
